File: api/core/src/main/kotlin/domain/Kontakt.kt
package domain

import base.Encrypted
import base.Id
import kotlinx.serialization.Contextual
import kotlinx.serialization.Serializable

@Serializable
data class Kontakt(
    override var _id: Id<Kontakt> = Id(),
    var oseba_id: MutableSet<Id<Oseba>>, // Mama in sin si lahko delita kontakt!
    var tip: Tip,
    @Contextual val data: Encrypted
) : Entiteta<Kontakt> {
    enum class Tip { EMAIL, TELEFON }
}

File: app/src/app/services/api/openapi/models/kontakt.ts
/* tslint:disable */
/* eslint-disable */
import { BsonBinary } from './bson-binary';
import { Id } from './id';
import { ObjectId } from './object-id';
export interface Kontakt {
  '_id'?: ObjectId;
  data?: BsonBinary;
  oseba_id?: Array<Id>;
  tip?: 'EMAIL' | 'TELEFON';
}
